src/serializer/ResidualReactElementSerializer.js

```diff
@@ -252,7 +252,24 @@ export class ResidualReactElementSerializer {
     if (reactLibraryObject !== undefined) {
       this.residualHeapSerializer.serializeValue(reactLibraryObject);
     }
-    let identifier = convertExpressionToJSXIdentifier(this.residualHeapSerializer.serializeValue(typeValue), true);
+    let identifier;
+    if (typeValue instanceof SymbolValue) {
+      // if there is no React library, then we should throw and error, as it is needed for React.Fragment output
+      if (reactLibraryObject === undefined) {
+        throw new FatalError("unable to serialize JSX fragment due to React not being referenced in scope");
+      }
+      // we want to vist the Symbol type, but we don't want to serialize it
```

Comment:
There are more things described as symbols. We shouldn't just assume that if it's a symbol, it must be a fragment. We should throw for unknown ones.

Reply:
Address in a new commit